build/bin/sage-logger

```diff
@@ -63,15 +63,15 @@ fi
 
 timefile="$logdir/$logname.time"
 rm -f "$timefile"
-if /usr/bin/time -h -o /dev/null true; then
+if /usr/bin/time -h -o /dev/null true 2>/dev/null; then
```

Comment:
I'm sure ubuntu has `/usr/bin/time` available. The problem is that you are using nonstandard arguments to `time(1)`:
```
$ /usr/bin/time -h -o /dev/null
/usr/bin/time: invalid option -- 'h'
Try '/usr/bin/time --help' for more information.
$ echo $?
125
```
Also, if you want to parse the time you should pass `-p` to ensure the standard format is used (https://www.man7.org/linux/man-pages/man1/time.1p.html), and then you can just use the bash builtin.

For what you are trying to acomplish, you could also use bash variables `$EPOCHSECONDS` (or `$EPOCHREALTIME` for more precision).

Reply:
The reviewer specifically asked for the human-readable format to be used.
The whole thing is just a convenience.

Reply:
> The reviewer specifically asked for the human-readable format to be used. The whole thing is just a convenience.

I don't know what is "the human-readable format" but `-h` is not supported by gnu `time(1)` nor by the bash builtin.

Gnu time does support `-o` but not `-h`.

If you want to bypass the bash builtin, the better approach is to use `\time` which will still search in `PATH`.

> For what you are trying to acomplish, you could also use bash variables $EPOCHSECONDS (or $EPOCHREALTIME for more precision).

There's also `$SECONDS`.

Reply:
> The reviewer specifically asked for the human-readable format to be used. The whole thing is just a convenience.

I see, but then the bash builtin time is just what the reviewer asked for.

You could
 - do the redirect `2>&1` inside the `sh -c "..."`
 - capture with `2>` the output of time

(since the bash builtin doesn't support -o)

Reply:
If you want to prepare a follow-up PR, I can test it for its portability. I don't have time to work on it myself at the moment.

Reply:
> If you want to prepare a follow-up PR, I can test it for its portability. I don't have time to work on it myself at the moment.

Please don't tell me what I should do with my time.

My comment before was just remarking that the current PR does not fix the issue but only hides it. Reverting cb4e7184b for the time being seems to be easier workaround, until you or someone else gets the time to improve the output in a portable way.

Reply:
> Please don't tell me what I should do with my time.

@tornaria And I am not, so don't tell me what I shouldn't tell you.